lib/Controller/LockingController.php

```diff
@@ -90,6 +93,12 @@ public function lockFolder(int $id): DataResponse {
 			throw new OCSForbiddenException($this->l10n->t('You are not allowed to create the lock'));
 		}
 
+		if ($userFolder->getId() === $id) {
+			$e = new OCSForbiddenException($this->l10n->t('You are not allowed to lock the root'));
+			$this->logger->error($e->getMessage(), ['exception' => $e])
+			throw $e;
```

Comment:
Can you also output OCSForbiddenException?

Right now curl returns with 200, while preventing lock.